File: sdk/client/javascript/openapi-test/test/klaytnDebug/vMTracing/DebugTraceBlockFromFile.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");
const {join} = require('path');

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('debug_traceBlockFromFile API', () => {
    test('should return debug_traceBlockFromFile', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            done();
        };

        const fileName = "/home/sotatek/block.rlp";

        sdk.debug.traceBlockFromFile(fileName, {}, callbackOne);
    });
});

File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/debug/vMTracing/DebugTraceBlockFromFileTest.java
package opensdk.sdk.apis.debug.vMTracing;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugTraceBlockFromFileResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

@DisplayName("Debug RPC Test")
public class DebugTraceBlockFromFileTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_traceBlockFromFile")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        String fileName = "/home/sotatek/block.rlp";

        DebugTraceBlockFromFileResponse response = sdk.debug.traceBlockFromFile(fileName, null).send();

        assertNotNull(response);
        assertNull(response.getError());
    }
}
